pipeline: buildprep from PVC for devel builds

jlebon opened the pull request:
So then devel pipelines can have incremental versions and test the logic
around no-ops too.

jlebon commented:
You'll need to clear out your PVC build dir first before testing this though, since it relies on how we now save builds in there.

dustymabe commented on Jenkinsfile:
```diff
@@ -100,6 +100,13 @@ podTemplate(cloud: 'openshift', label: 'coreos-assembler', yaml: pod, defaultCon
                 coreos-assembler buildprep s3://${s3_builddir}
                 """)
+            } else if (!prod && utils.path_exists("/srv/devel/${devel_prefix}/build")) {
+                // XXX: should integrate this into buildprep
```
i.e. buildprep could accept a file:/// URI?

jlebon replied:
Right, exactly. Would be trivial, just didn't want to distract away from getting pipelines in.

dustymabe commented on Jenkinsfile:
```diff
@@ -100,6 +100,13 @@ podTemplate(cloud: 'openshift', label: 'coreos-assembler', yaml: pod, defaultCon
+            } else if (!prod && utils.path_exists("/srv/devel/${devel_prefix}/build")) {
+                // XXX: should integrate this into buildprep
+                utils.shwrap("""
+                # really we only need some of the artifacts, but meh...
```
yeah I thought we only really needed the commit object and maybe the meta.json or something

dustymabe commented:
will test

jlebon commented:
This now needs https://github.com/coreos/coreos-assembler/pull/561.

dustymabe commented on Jenkinsfile:
```diff
@@ -181,9 +187,9 @@ podTemplate(cloud: 'openshift', label: 'coreos-assembler', yaml: pod, defaultCon
-              cp -a \$(realpath builds/latest) /srv/devel/${devel_prefix}/build
+              rm -rf ${devel_builddir}
+              mkdir -p ${devel_builddir}
+              cp -aT builds ${devel_builddir}
```
you made this change so we would include `builds.json` now?

jlebon replied:
Yup, exactly!

jlebon commented:
Dropped WIP on this. Tested working fine with https://github.com/coreos/coreos-assembler/pull/561.
I also added a `MINIMAL` parameter to make hacking on the pipeline faster.

dustymabe commented:
> Dropped WIP on this. Tested working fine with [coreos/coreos-assembler#561](https://github.com/coreos/coreos-assembler/pull/561).

It's working for me too. Merge away when ready. 

> I also added a `MINIMAL` parameter to make hacking on the pipeline faster.

+1 - that will help cut down on test times
